src/lib/bootloader/mbr_update.rb

```diff
@@ -142,20 +143,36 @@ def disks_to_rewrite
       ret.uniq
     end
 
+    MAX_BIOS_ID = 1000
     def first_base_device_to_boot(md_device)
-      md = Yast::BootStorage.Md2Partitions(md_device)
-      md.reduce do |res, items|
-        device, bios_id = items
-        next device unless res
-
-        bios_id < md[res] ? device : res
+      md = ::Bootloader::Stage1Device.new(md_device)
+      lowest_id = MAX_BIOS_ID
+      md.real_devices.reduce do |res, device|
+        res ||= device # ensure at least one device is used, even if none have bios_id
+        bios_id_num = bios_id_for(device)
+        if lowest_id > bios_id_num
+          lowest_id = bios_id_num
+          next device
+        else
+          res
+        end
```

Comment:
Could this be written more clearly as `md.real_devices.sort_by {|d| bios_id_for(d)}.first` ?